proposals/11765-KFP-SDK-common-component-support/11765-KFP-SDK-common-component-support.md

```diff
@@ -0,0 +1,100 @@
+
+These issues lead to inefficiencies, increased development time, and a less-than-ideal user experience; they ultimately get in the way of wider adoption and effective use of KFP for ML workflows.
+
+## Proposal: SDK Support for Common Components
+
+This proposal advocates for a fundamental shift in how KFP provides common component support: **integration directly into the KFP SDK.**  This involves:
+
+* **Deprecate and Replacing the [`pipelines/components`](../../components) Directory:**
+    * **Retire Outdated/POC Components:**  Identify and remove components in [`pipelines/components`](../../components) that are outdated, buggy, or lack sufficient quality.
+    * **Migrate Useful Examples:**  Some components appear to be meant more as proof-of-concepts, than as production-ready tools; these are better kept in the [`samples`](../../samples/) directory. Relocate valuable, proof-of-concept components to the [`samples`](../../samples/) directory as illustrative examples.
+    * **Discontinue Active Development:**  Cease further development and maintenance of the [`pipelines/components`](../../components) directory as the primary source of reusable components.
+
+* **Develope SDK-Integrated Component Categories:**  Organize core, widely applicable components within the KFP SDK under logical functional categories.  This proposal suggests the following initial categories (subject to review), mirroring common ML workflow stages:
+
+    * **`kfp.components.data`:**  Components for data ingestion, transformation, and validation. Conceptual examples:
+        * `load_from_gcs(...)`: Load data from Google Cloud Storage.
+        * `split_dataset(...)`: Split a dataset into training and validation sets.
+        * `validate_schema(...)`: Validate data against a schema.
+
+    * **`kfp.components.train`:** Components for model training and hyperparameter tuning.  Conceptual examples:
+        * **`kubeflow_trainer(...)`:**  Integration with Kubeflow Training Operators
+        * **`hyperparameter_tune(...)`:**  Integration with Katib for hyperparameter optimization.
+        * `xgboost.train(...)`:  Component for XGBoost training.
+
+    * **`kfp.components.serve`:** Components for model deployment and serving. Conceptual examples:
+        * **`deploy_kserve(...)`:** Integrate with KServe for model serving on Kubernetes.
+        * `register_model(...)`:  Register a model in a model registry.
+
+    * **`kfp.components.validate`:** Components for model evaluation and monitoring. Conceptual examples:
+        * `evaluate_classification_model(...)`:  Calculate standard evaluation metrics.
+        * `data_drift_detection(...)`: Detect data drift using statistical methods.
+
+Note that the conceptual examples here are for illustrative purposes, and are not meant to be the final say.
+
+* **Implementation Details:**
```

Comment:
Similar to my comments in #11766, we need clear owners and guidelines around CVE remediation.